### DISCUSSION TITLE
Gestures for modal (swipe from top to bottom)

### DISCUSSION THREAD

**Discussion by punksta** [software-mansion/react-native-screens#946] [Ideas]
I would like to be able to swipe down to go back when 

`stackPresentation`: 'modal' or 'fullScreenModal'

is used

**Comment by miguelSWE** [comment]
On iOS you can swipe down to dismiss a `modal`
![2021-05-28 18 02 23](https://user-images.githubusercontent.com/2905749/120020550-c5ae1880-bfe1-11eb-8e26-6a2aa5607baf.gif)

`fullScreenModal` doesn't allow you to swipe down to dismiss as according to the Human Interface Guidelines by Apple
https://developer.apple.com/design/human-interface-guidelines/ios/app-architecture/modality/

`The fullscreen presentation style covers the entire screen. The previous view is completely covered, minimizing visual distraction. People dismiss a full-screen modal view by tapping a button.`

The reason I found your question is just because I'm looking at implementing the fullscreen and supporting the swipe down on the Mixcloud app. However, this doesn't follow the Human Interface Guidelines, it's more of our own use-case.

**Reply by punksta** [reply]
Appstore and Photos and some popular apps like Spotify have swipes down to go back for full-screen screens. 
I don't think Human Interface Guidelines should be followed strictly. 

I am working on big full-screen context menu with sub-menus which are fullscreen screens themself. Something similar to Spotify's track menu.

**Reply by milesscherrer** [reply]
This would be amazing. Currently have to use @react-navigation/stack instead of @reactnavigation/native-stack due to this.

**Comment by WoLewicki** [comment]
@milesscherrer @punksta can you check if https://github.com/software-mansion/react-native-screens/pull/1260 is something you need? Be aware that you should use e.g. `slide_from_bottom` animation instead of modal stack presentation since these modals cannot be dismissed and it cannot be changed.
